test/store_unittest.cc

```diff
@@ -0,0 +1,229 @@
+  MonitoredResource mr("type", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_THROW(store.LookupResource("missing_id"), std::out_of_range);
+  EXPECT_EQ(mr, store.LookupResource("id"));
+}
+
+TEST_F(MetadataStoreTest, MultipleResourcesWithSingleIdsCorrectlyStored) {
+  MonitoredResource mr1("type1", {});
+  MonitoredResource mr2("type2", {});
+  store.UpdateResource({"id1"}, mr1);
+  store.UpdateResource({"id2"}, mr2);
+  EXPECT_EQ(mr2, store.LookupResource("id2"));
+  EXPECT_EQ(mr1, store.LookupResource("id1"));
+}
+
+TEST_F(MetadataStoreTest, SingleResourceWithMultipleIdsCorrectlyStored) {
+  MonitoredResource mr("type", {});
+  store.UpdateResource({"id1", "id2", "id3"}, mr);
+  EXPECT_EQ(mr, store.LookupResource("id3"));
+  EXPECT_EQ(mr, store.LookupResource("id1"));
+  EXPECT_EQ(mr, store.LookupResource("id2"));
+}
+
+TEST_F(MetadataStoreTest, MultipleResourcesAndIdsCorrectlyStored) {
+  MonitoredResource mr1("type1", {});
+  MonitoredResource mr2("type2", {});
+  store.UpdateResource({"id1", "id2", "id3"}, mr1);
+  store.UpdateResource({"id-a", "id-b", "id-c"}, mr2);
+  EXPECT_EQ(mr1, store.LookupResource("id1"));
+  EXPECT_EQ(mr1, store.LookupResource("id2"));
+  EXPECT_EQ(mr1, store.LookupResource("id3"));
+  EXPECT_EQ(mr2, store.LookupResource("id-a"));
+  EXPECT_EQ(mr2, store.LookupResource("id-b"));
+  EXPECT_EQ(mr2, store.LookupResource("id-c"));
+}
+
+TEST_F(MetadataStoreTest, ResourceToIdsAssociationCorrectlyUpdated) {
+  MonitoredResource mr("type", {});
+  store.UpdateResource({"id1", "id2"}, mr);
+  EXPECT_EQ(mr, store.LookupResource("id1"));
+  EXPECT_EQ(mr, store.LookupResource("id2"));
+  store.UpdateResource({"id-a", "id-b"}, mr);
+  EXPECT_EQ(mr, store.LookupResource("id1"));
+  EXPECT_EQ(mr, store.LookupResource("id2"));
+  EXPECT_EQ(mr, store.LookupResource("id-a"));
+  EXPECT_EQ(mr, store.LookupResource("id-b"));
+}
+
+TEST_F(MetadataStoreTest, DefaultMetadataMapIsEmpty) {
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_TRUE(metadata_map.empty());
+}
+
+TEST_F(MetadataStoreTest, UpdateResourceDoesNotUpdateMetadata) {
+  MonitoredResource mr("type", {});
+  store.UpdateResource({"id1"}, mr);
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_TRUE(metadata_map.empty());
+}
+
+TEST_F(MetadataStoreTest, UpdateMetadataChangesMetadataMap) {
+  MonitoredResource mr("type", {});
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr, std::move(m));
+  const auto metadata_map = GetMetadataMap();
+  EXPECT_EQ(1, metadata_map.size());
+  EXPECT_EQ("default-version", metadata_map.at(mr).version);
+}
+
+TEST_F(MetadataStoreTest, MultipleUpdateMetadataChangesMetadataMap) {
+  MonitoredResource mr1("type1", {});
+  MonitoredResource mr2("type2", {});
+  MetadataStore::Metadata m1(
```

Comment:
Observation, I got really confused when reading the comparisons, as I didn't pick up on mr1 and m1. Would it be worth using metadata1 and metadata2 to keep it incredibly obvious? I know it's not "consistent", but it would have helped when glancing through quickly.

Reply:
Done.

Reply:
@bmoyles0117 you're right about the confusion, but I actually prefer `m` over `metadata`, because `metadata.metadata` reads really weird. Can we instead address it by renaming `mr` to `resource`? Sorry, @dhrupadb...

Reply:
Done.